lib/create-app.js

```diff
@@ -200,6 +183,25 @@ function initWebId (argv, app, ldp) {
   // Set up authentication-related API endpoints and app.locals
   initAuthentication(app, argv)
 
+  // Protect against requests from third-party applications
+  app.use((req, res, next) => {
+    // Reject cookies from third-party applications.
+    // Otherwise, when a user is logged in to their Solid server,
+    // any third-party application could perform authenticated requests
+    // without permission by including the credentials set by the Solid server.
+    const origin = req.headers.origin
+    const userId = req.session.userId
+    if (!argv.host.allowsSessionFor(userId, origin)) {
+      debug(`Rejecting session for ${userId} from ${origin}`)
+      // Destroy session data
+      delete req.session.userId
+      // Ensure this modified session is not saved
+      req.session.save = done => done()
+    }
+    next()
+  })
```

Comment:
By moving the code after the auth handlers (but still before the LDP handlers), we ensure that third-party origins can log you out.